setup.py

```diff
@@ -71,31 +73,31 @@ def build_p2p_daemon():
         with tarfile.open(dest, "r:gz") as tar:
             tar.extractall(tempdir)
 
-        for executable in EXECUTABLES:
-            result = subprocess.run(
-                ["go", "build", "-o", os.path.join(here, "hivemind", "hivemind_cli", executable)],
-                cwd=os.path.join(tempdir, f"go-libp2p-daemon-{P2PD_VERSION.lstrip('v')}", executable),
-            )
-            if result.returncode != 0:
-                raise RuntimeError(f"Failed to build {executable}: exited with status code: {result.returncode}")
+        result = subprocess.run(
+            ["go", "build", "-o", os.path.join(here, "hivemind", "hivemind_cli", "p2pd")],
+            cwd=os.path.join(tempdir, f"go-libp2p-daemon-{P2PD_VERSION.lstrip('v')}", "p2pd"),
+        )
+        if result.returncode != 0:
+            raise RuntimeError(f"Failed to build p2pd: exited with status code: {result.returncode}")
 
 
 def download_p2p_daemon():
-    for executable, expected_hash in EXECUTABLES.items():
-        binary_path = os.path.join(here, "hivemind", "hivemind_cli", executable)
+    binary_path = os.path.join(here, "hivemind", "hivemind_cli", "p2pd")
+    os_name = platform.system().lower()
+    expected_hash = P2P_BINARY_HASH[os_name]
 
-        if sha256(binary_path) != expected_hash:
-            binary_url = os.path.join(P2PD_BINARY_URL, executable)
-            print(f"Downloading {binary_url}")
+    if sha256(binary_path) != expected_hash:
```

Comment:
Just to clarify: do we verify it this way to automatically download the correct binary for a non-Linux system? Would it be more clear if we simply checked that the binary exists?

Reply:
Sometimes devs replace the binary for testing purposes (e.g., to test an update in libp2p-daemon). As far as I understand, this code allows to replace the binary to the standard one when you reinstall hivemind (since the expected behavior in this case is to get a clean installation).